docs/types/resolvers.md

```diff
@@ -4,4 +4,205 @@ title: Resolvers
+
+
+@strawberry.type
+class User:
+    name: str
+
+
+@strawberry.type
+class Query:
+    @strawberry.field
+    def user(self, id: strawberry.ID) -> Optional[User]:
+        return User(name="Marco")
+---
+type User {
+    name: String!
+}
+
+type Query {
+    user(id: ID!): User
+}
+```
+
+## Accessing parent's data
+
+It is quite common to want to be able to access the data from the parent in a
+resolver, for example let's say that we want to defined a `fullName` field on
+our user. We can defined a new field with a resolver that combines the first
+name and last name:
+
+```python+schema
+import strawberry
+
+
+@strawberry.type
+class User:
+    first_name: str
+    last_name: str
+
+    @strawberry.field
+    def full_name(self) -> str:
+        return f"{self.first_name} {self.last_name}"
+---
+type User {
+    firstName: String!
+    lastName: String!
+    fullName: String!
+}
+```
+
+In the case of a decorated resolver you can use the _self_ parameter as you
+would do in a normal python class[^1].
+
+For resolvers defined as normal python functions you can use the special `root`
+parameter, when defined, Strawberry will pass to it the value of the parent:
+
+```python
+import strawberry
+
+def full_name(root: User) -> str:
+    return f"{root.first_name} {root.last_name}"
+
+@strawberry.type
+class User:
+    first_name: str
+    last_name: str
+    full_name: str = strawberry.field(resolver=full_name)
+```
+
+## Accessing execution information
+
+Sometimes it is useful to access the information for the current execution
+context, to do so you can provide the `info` parameter to resolvers, like this:
+
+```python
+import strawberry
+from strawberry.types import info
+
+def full_name(root: User, info: Info) -> str:
+    return f"{root.first_name} {root.last_name} {info.field_name}"
+
+@strawberry.type
+class User:
+    first_name: str
+    last_name: str
+    full_name: str = strawberry.field(resolver=full_name)
+```
+
+### API
+
+`Info[ContextType, RootValueType]`
+
+| Parameter name  | Type                       | Description                                                           |
+| --------------- | -------------------------- | --------------------------------------------------------------------- |
+| field_name      | `str`                      | The name of the current field                                         |
+| context         | `ContextType`              | The value of the context                                              |
+| root_value      | `RootValueType`            | The value for the root type                                           |
+| variable_values | `Optional[Dict[str, Any]]` | The variables for this operation                                      |
+| operation       | `OperationDefinitionNode`  | The ast for the current operation (public API might change in future) |
+| path            | `Path`                     | The path for the current field                                        |
```

Comment:
I'm a bit confused on what this is the API for? Where do these parameters reside?

Reply:
After looking at the other files, I see that `Info` is generic of types `ContextType` and `RootValueType`. I don't think this syntax here makes that clear

Reply:
do you have any suggestion, on to make it look nicer?

this is how it looks now: 
![image](https://user-images.githubusercontent.com/667029/107261411-929a4600-6a3f-11eb-8deb-4a74b6d5f8ac.png)

Reply:
I think it might just be best to make it clear that Info is Generic (I don't think just the `[]`s will make that clear) and that those parameters are for it's `__init__` method

As far as the first is concerned maybe just use
```python
class Info(Generic[ContextType, RootValueType]): ...
```

Reply:
yes! similar to the python docs: 
![image](https://user-images.githubusercontent.com/667029/107889768-18713200-6f15-11eb-82e9-5ac819879f33.png)

![image](https://user-images.githubusercontent.com/667029/107889887-d694bb80-6f15-11eb-9d4c-6587de6ccf6e.png)

Reply:
A difference is that in the Python docs, the `__init__` argslist is in the parens, but here we have the base/generic classes. I think this is probably fine for now, though